go/capship/commands/server.go

```diff
@@ -0,0 +1,159 @@
+/*
+Copyright 2018 The cap Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package commands
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/IBM/cap/go/log"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	alertsDirectory = "alerts/"
+	feedsDirectory  = "feeds/"
+	usFeed          = "cap.us"
+)
+
+// Server - capship host
+type Server struct {
+	config  *Config
+	context context.Context
+}
+
+// New - Allocates a new instance of a capship server
+func New(ctx context.Context, config *Config) (*Server, error) {
+	switch {
+	case config.Root == "":
+		return nil, errors.New("root must be specified")
+	}
+
+	if err := os.MkdirAll(config.Root+alertsDirectory, 0711); err != nil {
+		return nil, err
+	}
+	if err := os.MkdirAll(config.Root+feedsDirectory, 0711); err != nil {
+		return nil, err
+	}
+	var (
+		s = &Server{
+			config:  config,
+			context: ctx,
+		}
+	)
+	// set the log level for the server context
+	lvl, err := log.ParseLevel(config.LogLevel)
+	if err != nil {
+		return nil, err
+	}
+	log.G(s.context).Level = lvl
+	return s, nil
+}
+
+// serve configures the http handlers then listens for http requests
+func (s *Server) serve() {
+	// host requests
+	http.HandleFunc("/cap/", s.pullHandler)
+	http.HandleFunc("/upload", s.uploadHandler)
+	http.Handle("/alerts/", http.StripPrefix("/alerts", http.FileServer(http.Dir(s.config.Root+alertsDirectory))))
+	http.Handle("/feeds/", http.StripPrefix("/feeds", http.FileServer(http.Dir(s.config.Root+feedsDirectory))))
```

Comment:
Is this just feeds published by this server or does it include stuff that we add via the upload handler?